chainercv/links/model/feature_extraction_predictor.py

```diff
@@ -0,0 +1,137 @@
+import numpy as np
+import warnings
+
+import chainer
+from chainer import cuda
+
+from chainercv.transforms import center_crop
+from chainercv.transforms import scale
+from chainercv.transforms import ten_crop
+
+
+class FeatureExtractionPredictor(chainer.Chain):
+
+    """Wrapper class that adds predict method to a feature extraction model.
+
+    The :meth:`predict` takes three steps to make predictions.
+
+    1. Preprocess images
+    2. Forward the preprocessed images to the network
+    3. Average features in the case when ten-crop is used.
+
+    Example:
+
+        >>> from chainercv.links import VGG16
+        >>> from chainercv.links import FeatureExtractionPredictor
+        >>> base_model = VGG16()
+        >>> model = FeatureExtractionPredictor(base_model)
+        >>> prob = model.predict([img])
+        # Predicting multiple features
+        >>> model.extractor.feature_names = ['conv5_3', 'fc7']
+        >>> conv5_3, fc7 = model.predict([img])
+
+    """
+
+    def __init__(self, extractor,
+                 size=(224, 224), scale_size=256,
+                 do_ten_crop=False):
```

Comment:
There are other cropping techniques, such as 5-crop and 144-crops. How about taking a string `crop` instead of a bool `do_ten_crop`? If `crop == 'center'`, `center_crop` is used (default behaviour). if `crop == '10'`, ten crop is used. We can add `'5'` and `'144'` when they become needed.